hotfix

Nabarb opened the pull request:
hotfix:
fixed a bug in nigelObj.updatePaths. Now it checks Mask before attempting to get the files paths in order not to check empty channels and throwing related errors.

Also, removed a line in the sort FeaturesUI that was causing the 3d plot to reset every time spikes were moved around.

Sometimes I wonder if my comments make sense to anyone.

Nabarb committed:
hotfix

hotfix:
fixed a bug in nigelObj.updatePaths. Now it checks Mask before attempting to get the files paths in order not to check empty channels and throwing related errors.

Also, removed a line in the sort FeaturesUI that was causing the 3d plot to reset every time spikes were moved around.

Sometimes I wonder if my comments make sense to anyone.

Nabarb commented:
The first bugfix deals with a niche situation where an error was thrown when a block with some disabled channels was moved to a new location and therefore updatePaths was called eventually.

Nabarb commented:
Fixing a second bug related to this. I will update this pull request soon.

Nabarb committed:
hotfix

hotfix2, strikes back.

block.MatchChannelsID was indexing the wrong variable inside a for. fixed.

Sort.parseBlocks, added two helper functions for future changes

Nabarb commented:
Should be ok to merge now.

m053m716 commented on +nigeLab/@Block/matchChannelID.m:
```diff
@@ -47,7 +47,7 @@
    for iCh = 1:N % For each channel
       % Match the channel ID to the correct row (if it exists)
-      tmp = find(ismember(masterID,C(iCh,:),'rows'),1,'first');
+      tmp = find(ismember(C,masterID(iCh,:),'rows'),1,'first');
```
@Nabarb How confident are you that this is correct? We're assigning `tmp` to `masterIdx` array for a given channel and block. So, shouldn't it be the corresponding index "into" the master, which will be accessed by some combination of indexing variables that indicate the current channel and block (elsewhere)?

Nabarb replied:
You are right and that is what it's doing, I believe. it looks for the index of the ID inside each block and logs the info into tmp, which is exactly the crossreferencing indexing variable you're talking about.
ie the channel with ID 8 is in position 8 in master but in position 7 in the third block, let's say. masterIdx(8,3) will be 7, and masterIdx(whatever,3) will be nan where the third block has the missing channel. Afterwards we will use masterIdx(iCh) when looking for the channel iCh in all blocks.

m053m716 replied:
@Nabarb OK as long as we are keeping that part consistent everywhere. Can we make a documentation of how "master" vs "child" Channel ID should be indexed, somewhere? Maybe it already exists...

m053m716 commented on +nigeLab/@Sort/parseBlocks.m:
```diff
@@ -32,7 +32,8 @@
+   curCh = 0;
+   nCh = numel(blockObj(ii).Mask);
    for iCh = blockObj(ii).Mask
       blockObj(ii).checkSpikeFile(blockObj(ii).Mask);
```
Wait, for every channel it checks the spike file using the Mask, but how does it know what the current channel is?

Nabarb replied:
I actually did not touch this, I only added the two things at the end. You're right though. Both curCh and iCh are not used. It was taking insanely long to do the checks, maybe that's why. let me look into it

Nabarb replied:
Now should be better.

m053m716 reviewed (changes requested):
I haven't thoroughly gone through all the associated logic, just had those 2 questions to double-check with you. If you're sure it's good just go ahead and merge.

Nabarb committed:
hotfix3 - Rise of the Bugs

parseBlocks now is checking the SpikeFiles only one time. It was checking them nChannels times before.
